kratos/tests/test_variables.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import print_function, absolute_import, division
+
+import KratosMultiphysics
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+
+class TestVariables(KratosUnittest.TestCase):
+    def test_01_GetSourceVariable(self): # Tests with numbers to force the order
+        component_variable = KratosMultiphysics.VELOCITY_X
+        source_variable = KratosMultiphysics.VELOCITY
+        false_source_variable = KratosMultiphysics.DISPLACEMENT
+        self.assertEqual(component_variable.GetSourceVariable(), source_variable)
+        self.assertNotEqual(component_variable.GetSourceVariable(), false_source_variable)
+
+    def test_02_VariableComponentConstructor(self):
+        NEW_VARIABLE = KratosMultiphysics.Array1DVariable3("NEW_VARIABLE")
+
+        NEW_VARIABLE_X = KratosMultiphysics.Array1DComponentVariable("NEW_VARIABLE_X", "NEW_VARIABLE", 1)
+        NEW_VARIABLE_Y = KratosMultiphysics.Array1DComponentVariable("NEW_VARIABLE_Y", "NEW_VARIABLE", 2)
+        NEW_VARIABLE_Z = KratosMultiphysics.Array1DComponentVariable("NEW_VARIABLE_Z", "NEW_VARIABLE", 3)
+
+    def test_03_CreateVariable(self):
+        NEW_STRING_VARIABLE = KratosMultiphysics.StringVariable("NEW_STRING_VARIABLE")
+
+        NEW_TEST_BOOL_VARIABLE = KratosMultiphysics.BoolVariable("NEW_TEST_BOOL_VARIABLE")
+
+        NEW_INT_VARIABLE = KratosMultiphysics.IntegerVariable("NEW_INT_VARIABLE")
+
+        NEW_INT_VECTOR_VARIABLE = KratosMultiphysics.IntegerVectorVariable("NEW_INT_VECTOR_VARIABLE")
+
+        NEW_DOUBLE_VARIABLE = KratosMultiphysics.DoubleVariable("NEW_DOUBLE_VARIABLE")
+
+        NEW_DOUBLE_VECTOR_VARIABLE = KratosMultiphysics.VectorVariable("NEW_DOUBLE_VECTOR_VARIABLE")
+
+        NEW_Q_VARIABLE = KratosMultiphysics.DoubleQuaternionVariable("NEW_Q_VARIABLE")
+
+    @KratosUnittest.expectedFailure
+    def test_04_CreateDuplicateVariableFail(self):
+        NEW_STRING_VARIABLE = KratosMultiphysics.StringVariable("NEW_STRING_VARIABLE")
+
+    @KratosUnittest.expectedFailure
```

Comment:
I think since you test if the error is thrown correctly you need the variable for which you test for duplicates exists already (from a previous test)

In order to make the tests independent I suggest the following:
~~~py
def test_CreateDuplicateVariableFail(self):
    NEW_STRING_VARIABLE = KratosMultiphysics.StringVariable("NEW_STRING_VARIABLE") # this works
    #check that a meaningful error is thrown
    with self.assertRaisesRegex(RuntimeError, "The error message"):
        NEW_STRING_VARIABLE_2 = KratosMultiphysics.StringVariable("NEW_STRING_VARIABLE") # this doe NOT work!
~~~

Reply:
yes
I usually run it once to get the error message since the formatting might be a bit different

Reply:
so if the error message in .cpp file changes one has to change this test also. instead if we put KratosUnittest.expectedFailure then it is independent of the message.